liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -128,15 +177,15 @@ else if (locValue.startsWith("ftp:")) {
                     return true;
                 else {
                     NodeList apps = incDoc.getElementsByTagName("application");
-                    if (isFoundWebAppElement(apps))
+                    if (isFoundAppConfigFromAppList(apps))
                         return true;
                 }
             }
         }
         return bFound;
     }
     
-    private static boolean isFoundInConfigDropinsDir(String serverFilePath, File configDirectory) {
+    private static boolean isFoundAppConfigFromDropinsDir(String serverFilePath, File configDirectory, String fileName) {
```

Comment:
Created new ServerConfigDocument to hold configured apps locations. This will cover all your review comments.